doc/api/process.md

```diff
@@ -1996,6 +1996,129 @@ Will generate an object similar to:
   tz: '2016b' }
 ```
 
+## Access Control
+
+> Stability: 1 - Experimental
+
+### process.accessControl.apply(restrictions)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `restrictions` {Object} A set of access restrictions that will be applied
+  to the current Node.js instance. The format should be the same as the one
+  returned by [`process.accessControl.getCurrent()`][]. Omitted keys will
+  default to `true`, i.e. to retaining the relevant permissions.
+
+*Warning*: This does not provide a full isolation mechanism. Child processes,
+native addons, and other communication channels may be used to circumvent
+these measures, if they have been made available before the corresponding
+restrictions have been put into place.
+
+This API is recent and may not be complete.
+Please report bugs at https://github.com/nodejs/node/issues.
+
+Operations scheduled before this call are not undone or stopped.
+Features that were previously disabled through this call cannot be re-enabled.
+Child processes do not inherit these restrictions, whereas [`Worker`][]s do.
+
+The following code removes some permissions of the current Node.js instance:
+
+```js
+process.accessControl.apply({
+  childProcesses: false,
+  createWorkers: false,
+  fsRead: false,
+  fsWrite: false,
+  loadAddons: false,
+  setV8Flags: false
+});
+```
+
+Keys not listed in the object, or with values not set to `false`,
+are unaffected.
+
+See [`process.accessControl.getCurrent()`][] for a list of permissions.
+
+### process.accessControl.getCurrent()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Object} An object representing a set of permissions for the
+  current Node.js instance of the following structure:
+
+<!-- eslint-skip -->
+```js
+{ accessInspectorBindings: true,
+  childProcesses: true,
+  createWorkers: true,
+  fsRead: true,
+  fsWrite: true,
+  loadAddons: true,
+  modifyTraceEvents: true,
+  netConnectionless: true,
+  netIncoming: true,
+  netOutgoing: true,
+  setEnvironmentVariables: true,
+  setProcessAttrs: true,
+  setV8Flags: true,
+  signalProcesses: true }
+```
+
+Currently only boolean options are supported.
+
+In particular, these settings affect the following features of the Node.js API:
```

Comment:
I find `fsRead` and `fsWrite` a bit confusing, since there are already APIs of this name.

Since this is for access control, perhaps we should use the naming convention `<verb> [<preposition>] <noun>`, e.g. `writeToFs` and `readFromFs`. This would affect the `childProcesses`, the `net*`, and perhaps others.

Reply:
Hm, interesting … is the only reason for this that these names refer to specific functions? The current naming scheme seems a bit more useful if we have multiple keys that we want to appear as groups when sorted alphabetically (e.g. `net*`).

Reply:
> is the only reason for this that these names refer to specific functions?

In the specific case of `fsRead` and `fsWrite` it is confusing, yes.

But keeping naming consistent within an API helps users. Especially since you say:

```
Keys not listed in the object, or with values not set to `false`, are unaffected.
```

This means if I mis-type a name then I will have a false sense of security. I would rather have a little verbosity than a security issue.

> if we have multiple keys that we want to appear as groups

I imagine this being useful primarily for APIs in the same module. You might address that with an increasingly verbose naming convention `<module>_<verb>[<preposition>]<noun>`. Are there cases where you would want to group keys that are part of different modules?

Reply:
So … what’s the `noun` for `fs` then? :) `fs_writeFilesystem`? :smile: 

My gut still says that we want something a) short/succinct and b) with the module coming first.

Maybe `fsObserve`/`fsModify`?